src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/ScanResultToComponentListTransformer.java

```diff
@@ -38,9 +38,15 @@ public List<Component> transformScanResultToComponentList(List<DeveloperScansSca
     private List<Component> convertExternalIDsToComponentList(HashMap<String, ScanMetadata> componentIdWithMetadata) {
         List<Component> componentList = new ArrayList<>();
         try {
-            for (String gav : componentIdWithMetadata.keySet()) {
-                String[] parts = gav.split(":");
-                componentList.add(new Component(parts[0], parts[1], parts[2], getJsonObjectFromScanMetadata(componentIdWithMetadata.get(gav))));
+            for (String componentIdString : componentIdWithMetadata.keySet()) {
+                String[] parts;
+                if ((parts = componentIdString.split(":")).length == 3) {
+                    // For Maven and Gradle, the componentId is of the form "g:a:v"
+                    componentList.add(new Component(parts[0], parts[1], parts[2], getJsonObjectFromScanMetadata(componentIdWithMetadata.get(componentIdString))));
+                } else if ((parts = componentIdString.split("/")).length == 2) {
+                    // For NPM and NuGet, the componentId looks is of the form "a/v"
+                    componentList.add(new Component(null, parts[0], parts[1], getJsonObjectFromScanMetadata(componentIdWithMetadata.get(componentIdString))));
```

Comment:
A static builder in Component can abstract the internal logic here to build a Component with group and without group.
**Component class:**
public Component(String artifactId, String versionId, Json object) {
this.groupId = null;
...
}

public static Component buildWithGroup(String[] parts, Json object) {
return new Component(parts[0], parts[1], object);
}

public static Component buildWithGroup(String[] parts, Json object) {
return new Component(parts[0], parts[1], parts[2], object);
}

**In above method transformScanResultToComponentList(...):**
JsonObject jsonObject = getJsonObjectFromScanMetadata(componentIdWithMetadata.get(componentIdString);
if ((parts = componentIdString.split(":")).length == 3) {
                    // For Maven and Gradle, the componentId is of the form "g:a:v"
                    componentList.add(Component.buildWithGroup(parts, jsonObject));
} else if ((parts = componentIdString.split("/")).length == 2) {
                    // For NPM and NuGet, the componentId looks is of the form "a/v"
                    componentList.add(Component.buildWithoutGroup(parts, jsonObject));
}

Reply:
Has been added to IDETECT-3950 as tech debt as per offline discussion with Madhu